evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -126,6 +145,14 @@ class Pipeline(pipeline_class):
                                                    "delay_target": include_delayed_features},
                    "pipeline": {"gap": gap, "max_delay": max_delay}})
 
+    def mock_predict(df):
+        return pd.Series(range(200, 200 + df.shape[0]))
+
+    if isinstance(pl, TimeSeriesRegressionPipeline):
+        mock_regressor_predict.side_effect = mock_predict
+    else:
+        mock_classifier_predict.side_effect = mock_predict
+
     if only_use_y:
         pl.fit(None, y)
```

Comment:
How does this work? Looking at the code for `fit()`, it seems like X is necessary to pass in, since you do call `estimator.fit()` on the computed features of X?

Reply:
When `X` is `None` it's assumed that the pipeline will create some features from the `y` before passing to the estimator (typically `DelayedFeatureTransformer`). We're getting rid of this assumption in #1517, however, because some (future) estimators will naturally support the `X=None` case.

Reply:
Gotcha, that makes sense. Thanks!